### ISSUE TITLE
[Autocomplete][a11y][v1] Outstanding a11y issues

### ISSUE THREAD

**Issue by lwise** [Shopify/polaris-react#789]
### What is the issue?

Autocomplete's accessibility is an improvement over web, but still has outstanding issues.
Devon left an in-depth explanation of these issues on the original Autocomplete PR: #1902 , as well as on the original ComboBox PR: #1807 .

### Current State/TLDR

* Autocomplete is currently following the 1.1 spec with non-<kbd>space</kbd>-able checkboxes.
* VoiceOver describes multi-select options as being checkboxes.
* Safari/VO occasionally announces options as "dimmed" instead of their labels.
* Firefox/NVDA does not read options if there is an Action Button included.

---

#### Problems
* The checkboxes result in a conflict between different specs:
  * Checkboxes are supposed to be toggled with <kbd>space</kbd>.
  * You are supposed to be able to type space characters into the text input by pressing <kbd>space</kbd>.
  * You can not do both of these things at the same time.
  * VoiceOver announces each option as a checkbox, implying that you can use <kbd>space</kbd> to select them, when you can not.
#### Investigation
* Safari/VO and Chrome/VO will ignore `role="presentation"` on `<input type="checkbox">`.
* Adding `aria-hidden="true"` to the `Checkbox` inside `OptionList` seems to prevent VoiceOver from reading the entire option label, even though the label is not included within the markup marked as "hidden".
#### Problems
* Safari/VO reads the options one behind the actual selection. [This is a known bug.](https://bugs.webkit.org/show_bug.cgi?id=167671)
  * This makes VoiceOver useless in using the component.
#### Problems
* Safari/VO occasionally reads options as "dimmed" instead of their labels.
* Firefox/NVDA can't read options when an Action Button is added into the list.
* Safari-mobile/VO can't swipe to the Popover contents.
* The entire ComboBox is tab-focusable.
#### Investigation
* Something in `PopoverOverlay` seems to be causing the "dimmed" problem.
  * Replacing the `PopoverOverlay` with a `div` causes the text to be read normally.
  * Getting rid of the a `.measuring` selector gets rid of the "dimmed" problem, but causes all optiosn to be read twice.
  * The selector was setting a `display: block` style on the overlay's content.
  * This is a huge mystery.
* Our linter requires something with the role "combobox" to be focusable.
### Non-Combobox
* This still involves a `listbox`, `option`s, and `aria-activedescendant`, but no `combobox`.
#### Problems
* Safari/VO reads everything as "dimmed".
#### Investigation
* Chrome/VO reads this perfectly.

### Other things that might be useful to know
* Popover contents don't render exactly where you might expect in the DOM, potentially causing problems.
* In most cases, Safari/VO will only read options if they are "flat" in hierarchy. `OptionList` does not render things in a "flat" way, so a ton of `li` elements within it have a `role="presentation"` set onto them.

**Comment by dpersing** [comment]
Raising the priority on this since it ties closely into list filters work.

**Comment by dpersing** [comment]
This issue would benefit from pairing between a FED and an accessibility specialist.

**Comment by dleroux** [comment]
Autocomplete is currently being reworked as a part of the [AutoComplete component project](https://github.com/Shopify/polaris-react/projects/3). Once we address this problem in the reworked component, we’ll close this issue.

**Comment by chloerice** [comment]
Closed by https://github.com/Shopify/polaris-react/pull/3910
